### PR TITLE
Add an option to customize the htmlpurifier whitelists from the module settings

### PR DESCRIPTION
**Pull request by craxrev**
These are some changes to the module settings that can enhance the custom pages feature customization:
- Allowed HTML whitelist field
- Allowed CSS properties whitelist field

Applied some regex validation rules to this fields.

### CODE REVIEW

**Comment by bumi** [comment]
hmm... this is a good idea and generally it is cool to make things configurable and not have it hard-coded. also looks like very nicely developed! 

but: I am not sure if we really should add that code and complexity and if it provides enough value to the whole product. for any line of code we have to ask if it really makes it easier for teams to work on their social causes. 

even if things seem easy in the beginning I have the experience that complexity is hidden... 
users need to understand the feature and be able to use it; 
users will do strange things with features and break it; 
bugs are not obvious but hidden
the more code and exceptions we have the more complicate it is in the future to make changes
we actually also have to write automated tests for all the functions. 

also to prioritize I think it is good to create issues and plan them by priority together. 

and keep in mind: it's null till we ship it. - before the first landing pages actually have been created and used on production the feature is `null` :) - it is as much as nothing developed at all before we really push it to production and people use it. 
so we need to make sure to get features online as quickly and simple as possible. 

hope you understand my thinking? 
what do you think?

**Comment by craxrev** [comment]
I understand your thinking and I'm not quite sure about the complexity part, see as we are in a testing phase, what I've just made is just some tweaks for the module so that further test can be conducted without returning to the code (i.e. change some hardcoded lines), that's the only reason, not because users might find it handy.

also I believe that this is a testing feature, and such features should not be handled like a real ones and I strongly believe that most of this feature will not be fully implemented on later iterations on production if not all of it.

what I think I did wrong here is that I put it in new pr, this should've been committed directly to master or better, should've been made from the get go?

I fully get your concern about making sure features get pushed to production asap for testing & feedback, and I'm on your side on this one, features should be as simple as possible, and settings/tweaks/modifs will come along with users' feedbacks.

**Comment by bumi** [comment]
no it was very correct to make a PR. exactly to discuss code and such proposals. - pretty much everything must go through a PR. 

if you say the user does not need this feature then I even less understand the need for this change. 
after we have reviewed the whitelist how likely is it that we constantly need to update that list? 

and no matter if it is for the user or testing... any code needs to be managed and can probably break or have security issues. so I don't see any difference, and would even argue if the code is not for the user then it is less likely that we need it. 

and you see we discuss here about features and spend time before we even have tried to build the very first landing page on production.

**Comment by bumi** [comment]
@gdalyy can you review and merge it if you two think we should have it. 

and let's in the future make issues for anything we develop and progress with the issues and order that we have in the project board: https://github.com/Coinsence/meta/projects/1

File: models/ContainerPage.php [hunk 1]
```diff
@@ -120,10 +120,11 @@ public function reformatFilter($title)
 	 */
     public function purifyFilter($html)
     {
+	    $settings = new SettingsForm();
 	    $purifierConfig = [
-			    'HTML.Allowed' => '*[style],*[class],div,p,br,b,strong,i,em,u,s,a[href|target],ul,li,ol,span,h1,h2,h3,h4,h5,h6,sub,sup,blockquote,pre,img[src|alt],hr,font[size|color]',
+			    'HTML.Allowed' => $settings->htmlContainerPageAllowedHTML,
```
Comment (gdalyy) [inline models/ContainerPage.php]: You should fix fix indentation in $purifierConfig [ ]

**Review by gdalyy — CHANGES_REQUESTED** [review]

**Comment by gdalyy** [comment]
Otherwise lgtm :+1:

File: models/ContainerPage.php (hunk 1 above)
Comment (craxrev) [reply models/ContainerPage.php]: :+1:

**Comment by bumi** [comment]
@gdalyy when you approve your review, merge it.

**Review by gdalyy — APPROVED** [review]

**Pull request merged**